### DISCUSSION TITLE
Can swap function of useFieldArray move in all directions?

### DISCUSSION THREAD

**Discussion by Inhye-Cheong** [react-hook-form/react-hook-form#3093] [Q&A] [answered]
If I have mock data like below and using `swap` function in `useFieldArray`, Can I move first data (`id:0 data`) to last index (or like move last item to first item)?

```javascript
const mock = [
 {
   id: 0,
   name: 'hello1'
 },
{
   id: 1,
   name: 'hello2'
 },
{
   id: 2,
   name: 'hello3'
 },
{
   id: 3,
   name: 'hello4'
 },
];
```

If you have a good idea please comment thank you.

**Comment by bluebill1049** [comment, marked as answer]
perhaps you want to try `move`?

**Reply by Inhye-Cheong** [reply]
You are right. `swap` literally just swaps the positions of **from** and **to**. Two-way control works when I use `move` function. Thank you ! :)
